tools/nodejs_api/src_cpp/node_query_result.cpp

```diff
@@ -0,0 +1,107 @@
+#include "include/node_query_result.h"
+
+#include "include/all_async_worker.h"
+#include "include/each_single_async_worker.h"
+#include "main/kuzu.h"
+
+using namespace kuzu::main;
+
+Napi::FunctionReference NodeQueryResult::constructor;
+
+Napi::Object NodeQueryResult::Init(Napi::Env env, Napi::Object exports) {
+    Napi::HandleScope scope(env);
+
+    Napi::Function t = DefineClass(env, "NodeQueryResult", {
+       InstanceMethod("close", &NodeQueryResult::Close),
+       InstanceMethod("all", &NodeQueryResult::All),
+       InstanceMethod("getNext", &NodeQueryResult::GetNext),
+       InstanceMethod("hasNext", &NodeQueryResult::HasNext),
+    });
+
+    constructor = Napi::Persistent(t);
```

Comment:
Please double check if you need a persistent reference: https://github.com/nodejs/node-addon-api/blob/main/doc/object_reference.md